storage/flux/window.go

```diff
@@ -0,0 +1,183 @@
+package storageflux
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/arrow"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/values"
+	"github.com/influxdata/influxdb/v2"
+)
+
+// splitWindows will split a windowTable by creating a new table from each
+// row and modifying the group key to use the start and stop values from
+// that row.
+func splitWindows(ctx context.Context, in flux.Table, f func(t flux.Table) error) error {
```

Comment:
I had to trace this all the way back up the stack, and I realize that it's a common pattern, but I'd like to suggest a course correction going forward: this function injection pattern is used in a lot of places, and while the result may end up being less duplicated code, it comes at the cost of readable and maintainable code. This function just adds another layer to the stack. We have to open 6 different files between call point and function declaration.

I'm not asking for anything to be done in this patch. I am asking that we "walk, don't run" from this pattern in the future.

Reply:
This pattern has a lot to be desired and I agree I don't want it. My first implementation attempt was going to be to use the planner and inject a new transformation. Something like: `readAggregateWindow |> _splitWindows` if it were represented in flux code. I shied away from this implementation mostly for fear of either deadlocking the system or oversaturating it with tables. Because this has the potential for creating so many tables, I feared that it would overwhelm the dispatcher or potentially cause a deadlock for the same reason filter caused a deadlock. The split windows would be occupying a dispatcher slot and would not yield it to another transformation until it had split the entire table. I didn't view this as tenable and I'd like to revisit the execution engine to account for this because we're supposed to be able to account for that.

So this wasn't my preferred method and I'm happy to explore other avenues where we don't need to continuously wrap the table processing function.

Reply:
Right. I trust your judgment that it's the best solution for what we have at the time. I also appreciate your insight into other options you'd considered and (rightfully) chose to avoid. Mostly, I'd just like to start applying some friction to this pattern, so that we can start coming up with something that better lends itself to context-switching this logic regularly without too much cost.